### PR TITLE
[Central] - Multitable - Questions

### PR DESCRIPTION
**Pull request by drewjhart**
WIP

### CODE REVIEW

File: networking/src/ApiClient.ts [hunk 1]
```diff
@@ -885,10 +882,11 @@ class QuestionTemplateParser {
...
             instructions,
-            domain,
-            cluster,
-            grade,
-            standard,
+            answerSettings,
+            domain: domain ?? null,
+            cluster: cluster ?? null,
+            grade: grade ?? null,
+            standard: standard ?? null,
```
Comment (maniramezan) [inline networking/src/ApiClient.ts:889]: Is this for accounting for `undefined`?

File: networking/src/ApiClient.ts [hunk 2]
```diff
@@ -843,33 +832,41 @@ class QuestionTemplateParser {
     static questionTemplateFromAWSQuestionTemplate(
         awsQuestionTemplate: AWSQuestionTemplate
     ): IQuestionTemplate {
+        let gameTemplates: IGameTemplate[] = [];
+        if (!isNullOrUndefined(awsQuestionTemplate) && !isNullOrUndefined(awsQuestionTemplate.gameTemplates) && !isNullOrUndefined(awsQuestionTemplate.gameTemplates.items)) {
```
Comment (maniramezan) [inline networking/src/ApiClient.ts:836]: We should break this into multiple lines for readability

File: networking/src/ApiClient.ts [hunk 3]
```diff
@@ -780,7 +762,14 @@ class GameTemplateParser {
     static gameTemplateFromAWSGameTemplate(
         awsGameTemplate: AWSGameTemplate
     ): IGameTemplate {
-        let questionTemplates: IQuestionTemplate[] = awsGameTemplate.questionTemplates?.items?.filter((questionTemplate: AWSQuestionTemplate | null) => !isNullOrUndefined(questionTemplate)) || [];
+       let questionTemplates: IQuestionTemplate[] = [];
+        if (!isNullOrUndefined(awsGameTemplate) && !isNullOrUndefined(awsGameTemplate.questionTemplates) && !isNullOrUndefined(awsGameTemplate.questionTemplates.items)) {
```
Comment (maniramezan) [inline networking/src/ApiClient.ts:766]: Same as below

File: networking/src/ApiClient.ts [hunk 4]
```diff
@@ -684,12 +665,13 @@ type AWSQuestionTemplate = {
...
     choices?: string | null,
     instructions?: string | null,
+    answerSettings?: string | null,
     domain?: string | null | undefined,
     cluster?: string | null | undefined,
     grade?: string | null | undefined,
     standard?: string | null | undefined,
     imageUrl?: string | null | undefined,
-    gameTemplates?: IGameTemplate[] | null | undefined,
+    gameTemplates?:  IModelGameQuestionConnection | null,
```
Comment (maniramezan) [inline networking/src/ApiClient.ts:674]: nit: Extra space after `:`

File: networking/src/ApiClient.ts [hunk 5]
```diff
@@ -189,19 +164,25 @@ export class ApiClient implements IApiClient {
...
+            graphqlOperation(listQuestionTemplates, {limit: limit, nextToken })
         )) as { data: any }
         const parsedQuestionTemplates = result.data.listQuestionTemplates.items.map((questionTemplate: AWSQuestionTemplate) => {
             return QuestionTemplateParser.questionTemplateFromAWSQuestionTemplate(questionTemplate) as IQuestionTemplate
         });
         const parsedNextToken = result.data.listQuestionTemplates.nextToken;
         
         return { questionTemplates: parsedQuestionTemplates, nextToken: parsedNextToken };
+        } catch (e) {
+            console.log(e);
```
Comment (maniramezan) [inline networking/src/ApiClient.ts:183]: `console.error`?

File: networking/src/ApiClient.ts [hunk 6]
```diff
@@ -189,19 +164,25 @@ export class ApiClient implements IApiClient {
         ) {
             throw new Error(`Failed to create question template.`)
         }
-        return null; //QuestionTemplateParser.questionTemplateFromAWSQuestionTemplate(questionTemplate.data.createQuestionTemplate) as IQuestionTemplate
+        console.log(questionTemplate);
+        return null; // QuestionTemplateParser.questionTemplateFromAWSQuestionTemplate(questionTemplate.data.createQuestionTemplate) as IQuestionTemplate
```
Comment (maniramezan) [inline networking/src/ApiClient.ts:168]: intended?

File: networking/src/ApiClient.ts [hunk 7]
```diff
@@ -885,10 +882,11 @@ class QuestionTemplateParser {
             version,
             choices,
             instructions,
-            domain,
-            cluster,
-            grade,
-            standard,
+            answerSettings,
```
Comment (maniramezan) [inline networking/src/ApiClient.ts:885]: Since it's a JSON, should we cast it to a strongly type? Might be in another PR though

File: central/src/components/AddQuestionSidebar.jsx [hunk 8]
```diff
@@ -3,17 +3,56 @@ import { Route, Switch, useHistory } from 'react-router-dom';
...
+                    <Grid style={{height: 'calc(100vh - 64px)'}}>
+                      <p style={{color:"#797979", fontWeight:"bold"}}>No Game Selected</p>
+
+                      <h2 style={{width: "60%",color:"#797979", marginLeft:"auto", marginRight:"auto", marginTop:"30%"}}>In order to view questions, you must select a game from the section on the left</h2>
+                    </Grid>
+                  );
+                }
+              }/>
+            </Switch>           
+          </Grid> */}
```
Comment (maniramezan) [inline central/src/components/AddQuestionSidebar.jsx:43]: In case not used in another PR, we should remove it

File: central/src/components/CreateQuestion.jsx [hunk 9]
```diff
@@ -94,9 +94,30 @@ export default function QuestionForm({ updateQuestion, question: initialState, g
...
+    // if (questionToSend.id) {
+    //   newQuestion = await updateQuestion(questionToSend);
+    //   // we'll need to update this one as well
+    // } else {
+    //   newQuestion = await addQToGT(questionToSend);
+    //   // this is where we send to dynamoDB instead of RDS
+    //   delete newQuestion.updatedAt;
+    //   delete newQuestion.createdAt;
+    //   gameQuestion(newQuestion);
+    // }
```
Comment (maniramezan) [inline central/src/components/CreateQuestion.jsx:115]: Intentionally commented to be used in another PR?

File: central/src/components/CreateQuestion.jsx [hunk 10]
```diff
@@ -94,9 +94,30 @@ export default function QuestionForm({ updateQuestion, question: initialState, g
   // Handles grade, domain, cluster, or standard change/update
   const onSelectMaker = useCallback((field) => ({ target }) => { setQuestion({ ...question, [field]: target.value }); }, [question, setQuestion]);
 
+  const handleAddQTToGT = async (question) => {
+    console.log(addQToGT("a", "b"));
```
Comment (maniramezan) [inline central/src/components/CreateQuestion.jsx:98]: hmm, might worth to start adopting different layers of logging, `info` vs. `debug` vs. `error` vs. `warn` [reference](https://stackoverflow.com/a/36355027/1450348)

File: central/src/components/GameDashboard.jsx [hunk 11]
```diff
@@ -88,7 +88,7 @@ export default function GameDashboard({ loading, games, deleteGame, cloneGame, g
       return games
         .map((game, index) => {
           const { id, title, description, cluster, domain, grade, standard, imageUrl } = game;
-          const questionCount = game?.questions?.length || 0;
+          const questionCount = game?.questionTemplates?.length || 0;
```
Comment (maniramezan) [inline central/src/components/GameDashboard.jsx:91]: Instead of having `questionTemplates` to be nullable, should we just use an empty array when parsing server's response for it?

File: central/src/components/GameMaker.jsx [hunk 12]
```diff
@@ -362,6 +361,7 @@ export default function GameMaker({ loading, game, newSave, editSave, gameId, cl
               question={gameNumber ? null : getGameById(games, gameId)?.questions[Number(createQuestionIndex) - 1]}
               updateQuestion={updateQuestion}
               cloneQuestion={cloneQuestion}
+              addQToGT={addQToGT}
```
Comment (maniramezan) [inline central/src/components/GameMaker.jsx:364]: probably worth to have the full name, it's vague to know what this does without knowing the context

File: networking/amplify/backend/api/mobile/schema.graphql [hunk 13]
```diff
@@ -38,8 +38,9 @@ type QuestionTemplate @model {
   title: String!
   owner: String!
   version: Int!
-  choices: String
-  instructions: String
+  choices: AWSJSON
+  instructions: AWSJSON
+  answerSettings: AWSJSON
```
Comment (maniramezan) [inline networking/amplify/backend/api/mobile/schema.graphql:43]: Can this be nullable? Should we have a default value for it?

File: central/src/components/QuestionDashboard.tsx [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+type QuestionDashboardProps = {
+  questions: IQuestionTemplate[];
+};
+
+export default function QuestionDashboard({
+  questions
+}: QuestionDashboardProps) {
+  const classes = useStyles();
+  console.log("sup");
+  console.log(questions);
```
Comment (maniramezan) [inline central/src/components/QuestionDashboard.tsx:16]: probably only need one of these

File: central/src/containers/RouteContainer.tsx [hunk 15]
```diff
@@ -0,0 +1,298 @@
...
+      return true;
+  };
+
+  const filterGame = (game: IGameTemplate | null, search: string) => {
+    if (!game || !game?.title) {
+      return false;
+    }
+    if (game.title.toLowerCase().indexOf(search) > -1 || (game.description && game.description.toLowerCase().indexOf(search) > -1)) {
+      return true;
+    }
```
Comment (maniramezan) [inline central/src/containers/RouteContainer.tsx:242]: What's the purpose of this? If we're adding filtering for all games, we should make a server call to do so.

File: networking/src/ApiClient.ts (hunk 2 above)
Comment (drewjhart) [reply networking/src/ApiClient.ts:836]: fixed

File: networking/src/ApiClient.ts (hunk 3 above)
Comment (drewjhart) [reply networking/src/ApiClient.ts:766]: fixed

File: networking/src/ApiClient.ts (hunk 4 above)
Comment (drewjhart) [reply networking/src/ApiClient.ts:674]: fixed

File: networking/src/ApiClient.ts (hunk 5 above)
Comment (drewjhart) [reply networking/src/ApiClient.ts:183]: fixed

File: networking/src/ApiClient.ts (hunk 6 above)
Comment (drewjhart) [reply networking/src/ApiClient.ts:168]: no, fixed in a later PR

File: central/src/components/AddQuestionSidebar.jsx (hunk 8 above)
Comment (drewjhart) [reply central/src/components/AddQuestionSidebar.jsx:43]: removed

File: central/src/components/CreateQuestion.jsx (hunk 10 above)
Comment (drewjhart) [reply central/src/components/CreateQuestion.jsx:98]: good idea, I'll review and implement moving forward. This specific function has since been removed.

File: central/src/components/GameMaker.jsx (hunk 12 above)
Comment (drewjhart) [reply central/src/components/GameMaker.jsx:364]: fixed

File: networking/amplify/backend/api/mobile/schema.graphql (hunk 13 above)
Comment (drewjhart) [reply networking/amplify/backend/api/mobile/schema.graphql:43]: I've set values in the `graphql.schema` file and they are also initialized in `createQuestion`

File: central/src/components/QuestionDashboard.tsx (hunk 14 above)
Comment (drewjhart) [reply central/src/components/QuestionDashboard.tsx:16]: removed both in a later PR. Also, went through all the apps and pulled any needless console.logs from them as these kept coming up

File: central/src/containers/RouteContainer.tsx (hunk 15 above)
Comment (drewjhart) [reply central/src/containers/RouteContainer.tsx:242]: I think this is used to handle the search bar. This is old code that's been moved to a new container. Agree that it needs to change.

File: networking/src/ApiClient.ts (hunk 1 above)
Comment (drewjhart) [reply networking/src/ApiClient.ts:957]: `null` and `undefined` have been removed from the type per previous comments. Therefore `?? null` has been removed
